### PR TITLE
[integrations] go-metro

### PR DESCRIPTION
**Pull request by truthbk**
Add `go-metro` dependencies to build system.

### CODE REVIEW

File: package-scripts/datadog-agent/postinst [hunk 1]
```diff
@@ -78,6 +78,9 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
   chown -R dd-agent:root /usr/bin/dd-agent
   chmod 755 /usr/bin/dd-agent
 
+  # Configuring capabilities for DD-TCP-RTT
+  setcap cap_net_raw=eip /opt/datadog-agent/bin/dd-tcp-rtt
```
Comment (truthbk) [inline package-scripts/datadog-agent/postinst]: remove this - let customer run it manually.

File: config/projects/datadog-agent.rb [hunk 2]
```diff
@@ -83,8 +83,10 @@
   # SysVInit service file
   if redhat?
     extra_package_file '/etc/rc.d/init.d/datadog-agent'
+    runtime_dependency 'libcap2'
   else
     extra_package_file '/etc/init.d/datadog-agent'
+    runtime_dependency 'libcap2-bin'
```
Comment (olivielpeau) [inline config/projects/datadog-agent.rb]: You shouldn't declare `libcap` as a dependency of the `datadog-agent` package given that you embed the lib with `go-metro`. Tell me if I'm missing something though :)

File: omnibus_build.sh [hunk 3]
```diff
@@ -33,9 +33,10 @@ fi
 
 # Last but not least, let's make sure that we rebuild the agent everytime because
 # the extra package files are destroyed when the build container stops (we have
-# to tweak omnibus-git-cache directly for that). Same for gohai.
+# to tweak omnibus-git-cache directly for that). Same for gohai and rtt.
 git --git-dir=/var/cache/omnibus/cache/git_cache/opt/datadog-agent tag -d `git --git-dir=/var/cache/omnibus/cache/git_cache/opt/datadog-agent tag -l | grep datadog-agent`
 git --git-dir=/var/cache/omnibus/cache/git_cache/opt/datadog-agent tag -d `git --git-dir=/var/cache/omnibus/cache/git_cache/opt/datadog-agent tag -l | grep datadog-gohai`
+git --git-dir=/var/cache/omnibus/cache/git_cache/opt/datadog-agent tag -d `git --git-dir=/var/cache/omnibus/cache/git_cache/opt/datadog-agent tag -l | grep datadog-rtt`
```
Comment (olivielpeau) [inline omnibus_build.sh]: This should be updated to `datadog-metro` ;)

File: config/projects/datadog-agent.rb (hunk 2 above)
Comment (truthbk) [inline config/projects/datadog-agent.rb]: Hehe... yeah, it's not `libpcap` but `libcap` - this is the `capabilities` library that would be needed to set `CAP_NET_RAW` and allow traffic sniffing without being root. So this isn't at all PCAP related.

That said, we could make it a **soft** dependency and just instruct the customer to install it if they want to enable `go-metro`. Might be the safer course of action.

Comment (olivielpeau) [inline config/projects/datadog-agent.rb]: Got it, thanks for the explanation. I agree that we shouldn't have `libcap` as a required dependency though.

File: package-scripts/datadog-agent/postinst (hunk 1 above)
Comment (olivielpeau) [inline package-scripts/datadog-agent/postinst]: the command is documented in the agent now, let's delete it here ;)

File: config/projects/datadog-agent.rb (hunk 2 above)
Comment (olivielpeau) [inline config/projects/datadog-agent.rb]: Let's remove this runtime dependency for both platforms now that you've documented them in go-metro's example yaml conf.

File: omnibus_build.sh [hunk 4]
```diff
@@ -33,9 +33,10 @@ fi
 
 # Last but not least, let's make sure that we rebuild the agent everytime because
 # the extra package files are destroyed when the build container stops (we have
-# to tweak omnibus-git-cache directly for that). Same for gohai.
+# to tweak omnibus-git-cache directly for that). Same for gohai and rtt.
```
Comment (olivielpeau) [inline omnibus_build.sh]: nitpick: `rtt` -> `go-metro`

**Comment by olivielpeau** [comment]
Just one nitpick, other than that let's :shipit:

**Pull request merged**